0.4.1 releasenotes

j-wags opened the pull request:
Docs branch built at: https://open-forcefield-toolkit.readthedocs.io/en/041-releasenotes/releasehistory.html

- [x] #371 merged
- [x] Update [changelog](https://github.com/openforcefield/openforcefield/blob/master/docs/releasehistory.rst)

jchodera commented on docs/releasehistory.rst:
```diff
@@ -11,25 +11,72 @@ Releases follow the ``major.minor.micro`` scheme recommended by `PEP440 <https:/
 Current development
```
Should this be 0.4.1?

jchodera commented on docs/releasehistory.rst:
```diff
@@ -11,25 +11,72 @@ Releases follow the ``major.minor.micro`` scheme recommended by `PEP440 <https:/
 Current development
 -------------------
 
```
Do you need release summary text here?

j-wags commented:
@jchodera I've addressed your points above and make a few other fixes. Please take another look when possible. If it looks good I'll take care of the rest of the release.

j-wags commented:
Thanks -- The version number and description were total omissions on my part. I need to update the release checklist with this stuff -- Both to keep me on track when there's a lot going on, and to improve my bus factor :-)
